File: lab9-10/ex1/app/src/main/java/com/example/batterylevel/MainActivity.java
package com.example.batterylevel;

import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    BatteryLevel nivelBaterie;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        nivelBaterie = new BatteryLevel();
        IntentFilter filtru = new IntentFilter();
        filtru.addAction(Intent.ACTION_BATTERY_CHANGED);
        filtru.addAction(Intent.ACTION_BATTERY_LOW);
        registerReceiver(nivelBaterie, filtru);
    }
        @Override
        protected void onDestroy() {
            super.onDestroy();
            unregisterReceiver(nivelBaterie);
        }

}

File: lab7-8/Ex1/exercitiul1/app/src/main/java/com/example/intents/MainActivity.kt
package com.example.intents
import android.content.Intent
import android.os.Bundle
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Button
import androidx.compose.material3.Text
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.compose.ui.unit.dp

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        enableEdgeToEdge()
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        setContent{

            Column(
                modifier = Modifier.fillMaxSize().padding(top = 100.dp),
                verticalArrangement = Arrangement.Center,
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                Button(onClick = {
                       Intent(applicationContext, SecondActivity::class.java).also {
                    startActivity(it)
                }

                }) {
                    Text(text = "Acceseaza primul intent!")
                }


                Button(onClick = {
                    val intent = Intent(Intent.ACTION_SEND).apply {
                        type = "text/plain"
                        putExtra(Intent.EXTRA_EMAIL, arrayOf("dariusbabuteu@gmail.com"));
                        putExtra(Intent.EXTRA_SUBJECT, "Test email");
                        putExtra(Intent.EXTRA_TEXT, "Continutul emailului");
                    }
                    if (intent.resolveActivity(packageManager) != null) {
                        startActivity(intent)
                    }
                }) {
                    Text("Acceseaza al doilea intent!")
                }

                Button(onClick = {
                    Intent(Intent.ACTION_MAIN).also {
                        it.`package` = "com.google.android.youtube"
                        startActivity(it)
                    }
                })
                    {
                    Text("Acceseaza al treilea intent")
                }

            }

            }
        }
    }
